discovery-infra/test_infra/helper_classes/kube_helpers/base_resource.py

```diff
@@ -1,15 +1,32 @@
 import abc
 
-from .common import ObjectReference
+from .common import KubeAPIContext, ObjectReference
 
 
-class BaseCustomResource(abc.ABC):
+class BaseResource(abc.ABC):
+    """
+    A base object of both custom resources and kubernetes resources that holds
+    a shared KubeAPIContext. Any sub instance of this class will be added to the
+    shared context.
+    """
+    context = KubeAPIContext()
+
+    def __init__(self):
+        self.context.resources.add(self)
+
+    @abc.abstractmethod
+    def delete(self):
+        pass
+
+
+class BaseCustomResource(BaseResource):
     """
     Base class for all CRDs, enforces basic methods that every resource must
     have e.g create, path, get, delete and status.
     """
 
     def __init__(self, name: str, namespace: str):
+        BaseResource.__init__(self)
```

Comment:
use ``super()``

Reply:
why?

Reply:
easier refactoring of class name + shorter

Reply:
+ more python and more standard from what I know